src/ECDSARegistryCoordinator.sol

```diff
@@ -0,0 +1,867 @@
+        /**
+         * Get bitmap of quorums to register for and operator's current bitmap. Validate that:
+         * - we're trying to register for at least 1 quorum
+         * - the operator is not currently registered for any quorums we're registering for
+         * Then, calculate the operator's new bitmap after registration
+         */
+        uint256 quorumsToAdd = uint256(BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers, quorumCount));
+        uint256 currentBitmap = _currentOperatorBitmap(operatorId);
+        require(!quorumsToAdd.isEmpty(), "RegistryCoordinator._registerOperator: bitmap cannot be 0");
+        require(_quorumsAllExist(quorumsToAdd), "RegistryCoordinator._registerOperator: some quorums do not exist");
+        require(quorumsToAdd.noBitsInCommon(currentBitmap), "RegistryCoordinator._registerOperator: operator already registered for some quorums being registered for");
+        uint256 newBitmap = uint256(currentBitmap.plus(quorumsToAdd));
+
+        /**
+         * Update operator's bitmap, socket, and status. Only update operatorInfo if needed:
+         * if we're `REGISTERED`, the operatorId and status are already correct.
+         */
+        _updateOperatorBitmap({
+            operatorId: operatorId,
+            newBitmap: newBitmap
+        });
+
+        emit OperatorSocketUpdate(operatorId, socket);
+
+        if (_operatorInfo[operator].status != OperatorStatus.REGISTERED) {
+            _operatorInfo[operator] = OperatorInfo({
+                operatorId: operatorId,
+                status: OperatorStatus.REGISTERED
+            });
+
+            // Register the operator with the EigenLayer via this AVS's ServiceManager
+            serviceManager.registerOperatorToAVS(operator, operatorSignature);
+
+            emit OperatorRegistered(operator, operatorId);
+        }
+
+        /**
+         * Register the operator with the BLSApkRegistry, StakeRegistry, and IndexRegistry
+         */
+        (results.operatorStakes, results.totalStakes) = 
+            stakeRegistry.registerOperator(operator, operatorId, quorumNumbers);
+        results.numOperatorsPerQuorum = new uint32[](quorumNumbers.length);
+        for (uint256 i = 0; i < quorumNumbers.length; ++i) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            uint32 newTotalOperatorsForQuorum = totalOperatorsForQuorum[quorumNumber] + 1;
+            totalOperatorsForQuorum[quorumNumber] = newTotalOperatorsForQuorum;
+            results.numOperatorsPerQuorum[i] = newTotalOperatorsForQuorum;
+        }
+
+        return results;
+    }
+
+    function _getOrCreateOperatorId(
+        address operator,
+        ECDSAPubkeyRegistrationParams memory params
+    ) internal returns (bytes32 operatorId) {
+        operatorId = _operatorInfo[operator].operatorId;
+        if (operatorId == 0) {
+            operatorId = _registerECDSAPublicKey(operator, params, pubkeyRegistrationMessageHash(operator));
```

Comment:
I think this is a bug, and should be `pubkeyRegistrationMessageHash(params.signingAddress)` since this is verifying a signature against the params address, not the operator address. Can you certify whether I'm correct @ChaoticWalrus , need this for my branch as well.

Btw to differentiate the 2 addresses in ecdsa inc-sq, I've been using `operatorEcdsaAddress`, and `avsEcdsaAddress`. Perhaps we should stick to `operatorId` instead of `avsEcdsaAddress`, but it feels easier at least for the offchain code to prevent bugs by calling them the `operator` key and the `avs` key (meaning the key that is used to sign msgs offchain).

Reply:
hmmm nvm I think I'm wrong actually. Somehow the checkSignature is failing for me saying I have the wrong signer... let me double check.

Reply:
idk on naming. IMO depends on your intended usage.
I think `operatorId` is more general, but `avsEcdsaAddress` is more specific. Probably needs `ECDSA` in there somewhere, assuming you need different handlers for these things

Reply:
I don't know if I had thought the details of this through, but this method only gets run when called by the operator, and if the operator chooses a bad `signingAddress` that _could_ just be placed on them.
The signature here I guess is ultimately just confirming that the `signingAddress` "agrees" to the arrangement.

Reply:
One more thought on naming: I'll note as well that since we're using EIP1271 signature checking here, the `signingAddress` could be a contract that does whatever custom signature checking the operator wants, so it doesn't _strictly_ need to be ECDSA

Reply:
Super helpful comments! Will resolve nonetheless since my first point was just wrong.